aviary/subsystems/propulsion/propeller/propeller_map.py

```diff
@@ -105,13 +96,26 @@ def read_and_set_mach_type(self, data_file):
         # mach_type = mach
         # Hamilton Standard 10 Bladed Propfan Performance Deck:  Ct Tables
 
+        m_type = 'mach'  # default to freestream Mach number
+        m_type_define = False
         fp = get_path(data_file)
         with open(fp, "r") as f:
             line = f.readline()
-            line = f.readline()
-            line = f.readline()
-            line = f.readline()  # read 4th line
-            m_type = line.split()[3]  # get token 3
+            while len(line.split('#')[0].strip()) == 0:
+                line = f.readline()
+                try:
+                    substr = line.split('#')[1].split('=')[0].strip()
+                    if substr == 'mach_type':
+                        m_type = line.split('#')[1].split('=')[-1].strip()
+                        m_type_define = True
+                        break
+                except:
+                    pass
```

Comment:
```suggestion
            while len(line.split('#')[0].strip()) == 0:
                line = f.readline()
            if line.split('#')[0].split('=')[0].strip() == 'mach_type':
                m_type = line.split('#')[0].split('=')[-1].strip()
                m_type_define = True
```

Reply:
@crecine can you look at this branch:

[https://github.com/xjjiang/om-Aviary/tree/use_prop_table_csv](https://github.com/xjjiang/om-Aviary/tree/use_prop_table_csv)

I am not sure if the way I modify `csv_data_file.py` is good, but I must touch it one way or another if I adopt your suggestion of `mach_type = mach` or `mach_type, mach`.

Reply:
@crecine Perhaps we can close this PR and I open another PR for `mach_type` work?

Reply:
If Eliot is and Jason are fine with this as is, we can merge this PR and make the change to move required data out of comments as a follow on

Reply:
@jkirk5 , I requested @ehariton to review. The final issue is that the propeller map file has a parameter `mach_type = *` which is currently placed within comment block like the following:

```
# created 07/09/24 at 10:26
# GASP propeller map converted from general_aviation.map
# CT = f(Helical Mach at 75% Radius, Adv ratio & CP)
# mach_type = helical_mach
```

Carl suggested that `mach_type = helical_mach` be moved out of comment like the following:

```
# created 07/09/24 at 10:26
# GASP propeller map converted from general_aviation.map
# CT = f(Helical Mach at 75% Radius, Adv ratio & CP)
mach_type = helical_mach
```

But this requires that `utils/csv_data_file.py` be updated. I have an update:

[https://github.com/xjjiang/om-Aviary/tree/use_prop_table_csv](https://github.com/xjjiang/om-Aviary/tree/use_prop_table_csv)

but I don't really like it. So, I didn't add it to this PR. So, I would like to hear from both of you: should we close this PR as it is or make the change that Carl wants?

Reply:
I don't want to uncomment this line, for a very simple reason - it makes the file not a proper csv anymore. That would cause issues with the simple ways we can extract and save the data through `np.savetxt()` and similar functions.

Part of the reason I switched Aviary over to this format in the first place is to avoid the things that bothered me about FLOPS and GASP data files, primarily being lack of labeling and arbitrary handling of certain lines or columns.

Reply:
Now, I rewrite the code so that mach type is read from header instead of comment. I still keep `# mach_type = *` in the comment but it is not used. File `utils/csv_data_file.py` is not changed.

For `general_aviation.prop`, the header is:

```
    J, Helical_Mach,         CP,         CT
```
For `PropFan.prop`, the header is:
```
  J,       Mach,         CP,         CT
```

Reply:
that's an even better solution, I like it!

Reply:
I like this solution!